nixos/doc: use shallow-clone

06kellyjac opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
nixpkgs is big and some users have bad internet or slow disks so cloning
is very slow.

Could also put a branch if we want people to grab the release or at least unstable so they're not building off master with cache misses and breakages

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Ma27 commented:
In some cases it may not be desirable to do this. I think it'd be better to leave a small note what `--depth 1` does and why it *can* be useful.

06kellyjac commented:
done. Should I also add --branch so they use unstable or a release?
or at least a comment explaining branch?

06kellyjac committed:
nixos/doc: use shallow-clone

nixpkgs is big and some users have bad internet or slow disks so cloning
is very slow.

06kellyjac commented on nixos/doc/manual/from_md/installation/building-nixos.chapter.xml:
```diff
@@ -34,7 +34,9 @@
     <title>Practical Instructions</title>
     <programlisting>
-$ git clone https://github.com/NixOS/nixpkgs.git
+# a depth of 1 will result in a much faster download and setup of nixpkgs
+# but if you plan to do more advanced git operations you should consider removing it
+$ git clone https://github.com/NixOS/nixpkgs.git --depth 1
```
This specific documentation is for making your own isos & I've seen many people complaining how long clones take. People who don't need `--depth` know they don't need it and the description also helps

drupol commented on nixos/doc/manual/from_md/installation/building-nixos.chapter.xml:
```diff
@@ -34,7 +34,9 @@
     <title>Practical Instructions</title>
     <programlisting>
-$ git clone https://github.com/NixOS/nixpkgs.git
+# a depth of 1 will result in a much faster download and setup of nixpkgs
```
```suggestion
# The `depth` parameter in Git is used when you want to create a shallow clone of a repository. Shallow clone means downloading less history from the repository, thereby reducing the amount of data that needs to be transferred, which can significantly speed up the cloning process.
# For example, if you use git clone --depth 1, you're asking Git to only pull down one commit worth of historical data. This means you get the latest version of the codebase, but none of the history, which can be useful in cases where the history is large and you don't need it. It's quicker because you're simply downloading less data.
# However, note that this is a trade-off. While you gain speed and save disk space, you lose the repository's history and some functionalities like viewing log history beyond the shallow depth, searching for changes, and certain kinds of advanced Git operations.
```

drupol reviewed (changes requested):
I made some changes to the message. I think it's important to explain what's the use of the `depth` parameter.

drupol reviewed (changes requested):
What's the status of the PR ?

Do you plan to update it to get rid of the merge conflict ?
